test/e2e_node/device-plugin.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pborman/uuid"
+)
+
+const (
+	testPodNamePrefix = "nvidia-gpu-"
+	// Nvidia driver installation can take upwards of 5 minutes.
+	driverInstallTimeout = 10 * time.Minute
+)
+
+var _ = framework.KubeDescribe("[Feature:GPUDevicePlugin]", func() {
+	f := framework.NewDefaultFramework("device-plugin-gpus-errors")
+
+	// Install NVIDIA driver if cluster is running COS
+	BeforeEach(func() {
+		if !framework.IsNodeRunningCOS(getLocalNode(f)) {
+			Skip("NVIDIA GPU tests are supported only on Container Optimized OS image currently")
```

Comment:
What I was saying before is I *do not* think it is wise to add this gate to an e2e test.  Folks will want to enable this feature outside of GCE/GKE and we can warn, but we should not gate on it. 

/cc @ConnorDoyle @derekwaynecarr @davidopp @jeremyeder

Reply:
In that case the problem is not related to this PR only but all the GPU related tests (which are already merged in master) and I suggest we open an issue to discuss this as the scope of this discussion (which seem to be installing NVIDIA drivers from a container) is extremely vast.

That would allow this PR to continue being merged as it is doing the same kind of operations that are already in the code base (unless you consider this a blocker) :)

Does this seem like a reasonable course of action @timothysc ?

Reply:
Hmm, test dependence on a single distro seems like a blocker to me.

Reply:
Currently NVIDIA does not officially support installing the NVIDIA drivers through containers (COS is a work in progress).

For 1.8 we are trying to e2e test the device plugin on one of it's target hardware (GPUs)  As mentioned above we don't have a unified or supported solution for installing the NVIDIA driver.
So right now we are limiting ourselves to GKE using Google's GPU device plugin + driver installer.

Note that tests running specifically on COS have already been in the code for 5 months see:
https://github.com/kubernetes/kubernetes/pull/45136

Reply:
Instead of calling this file device-plugin.go, can we just call it gpu-device-plugin.go to indicate its current scope? After 1.8, we may want to replace these specific functions with more generic interface functions, e.g., instead of IsNodeRunningCOS, we can just use device-plugin-framework.NodeOSChecking().